### ISSUE TITLE
Span Components

### ISSUE THREAD

**Issue by azaroth42** [cheshire3/cheshire3#24]
Components which are generated from SpanXPaths do not generate the correct document from the documentStream.

I believe that this does the right thing, in ComponentDocumentStream, or at least a reasonable approximation of the right thing :)

```
    for src in self.sources:
        raw = src.process_record(session, rec)
        for xp in raw:
            if len(xp) == 2 and xp[0].__class__ == etree._Element:
                # span
                r = xp[0]
                doc = [etree.tostring(r)]
                for x in r.itersiblings():
                    if x == xp[1]:
                        break
                    else:
                        doc.append(etree.tostring(x))
                docstr = ''.join(doc)
                tree = r.getroottree()
                path = tree.getpath(r)
                docstr = """<c3component parent="%r" xpath="%s">%s</c3component>""" % (rec, path, docstr)
                doc = StringDocument(docstr)
                yield doc
                continue
       for r in xp: 
              ....
```

**Comment by bloomonkey** [comment]
I'm slightly concerned that the logic above could potentially give an wrong result whenever there are 2 matches, whether due to a SpanSelector or otherwise.

Will try to construct some unittests to ensure correct behaviour under all conditions.

**Comment by bloomonkey** [comment]
Fixed in commit 0ee70deed7a1303371f094c889235e0ae07282f4
